src/node-explorer-provider.ts

```diff
@@ -49,33 +49,78 @@ export class NodeExplorerProvider
     let transferItem = sources.get('application/vnd.code.tree.tsFileEntry');
     if (transferItem && transferItem.value) {
       const source: FileExplorer = transferItem.value[0];
-      return this.transferFile(source.uri, target);
+      return this.renameFile(source.uri, target);
     }
 
     transferItem = sources.get('text/uri-list');
     if (transferItem && transferItem.value) {
-      return this.transferFile(vscode.Uri.parse(transferItem.value), target);
+      transferItem.value.split('\r\n').forEach((uri: string) => {
+        return this.transferFile(vscode.Uri.parse(uri), target);
+      });
     }
+
+    sources.forEach(({ value }, mimeType) => {
+      console.log('mimeType', mimeType);
+
+      switch (mimeType) {
+        case 'application/vnd.code.tree.tsFileEntry':
+          return this.transferFile(value.uri, target);
+
+        // From the @types/vscode package: text/uri-list
+        //   A string with `toString()`ed Uris separated by `\r\n`.
+        case 'text/uri-list':
+          value.split('\r\n').forEach((uri: string) => {
+            return this.transferFile(vscode.Uri.parse(uri), target);
+          });
+          return;
+
+        default:
+          return;
+      }
+    });
   }
 
-  async transferFile(uri: vscode.Uri, target: FileExplorer) {
+  async renameFile(source: vscode.Uri, target: FileExplorer) {
+    const { address: sourceAddr } = parseTsUri(source);
+    const { address: targetAddr } = parseTsUri(target.uri);
+
+    if (sourceAddr !== targetAddr) {
+      // move across nodes
+      return vscode.window
+        .showErrorMessage('Moving files across nodes is not yet supported.', 'Open issue')
```

Comment:
Although slow, the current PR does support it. Any reason for turning it off?

Reply:
I didn't realize it did, as I stopped using `vscode.workspace.fs.readFile` for local files as it didn't support dropping from outside the workspace (dragging from Finder). I have updated it to also support node-to-node moving, while being a little more efficient when uploading directly from the local machine (using sftp.fastPut)